<a href="https://colab.research.google.com/github/fmakkawi/Data-205/blob/main/DATA205_Spring2026_Homework6_Farah_Makkawi.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files
uploaded = files.upload()

Saving athlete_events (1).csv to athlete_events (1).csv


In [2]:
from google.colab import files
uploaded = files.upload()

Saving host_city_locations.csv to host_city_locations.csv


In [3]:
import pandas as pd
import sqlite3

In [4]:
athlete_events = pd.read_csv('athlete_events (1).csv')
host_city_locations = pd.read_csv('host_city_locations.csv')

athlete_events.head()

,ID,Name,Sex,Age,Height,Weight,Team,NOC,Games,Year,Season,City,Sport,Event,Medal
0,1,A Dijiang,M,24.0,180.0,80.0,China,CHN,1992 Summer,1992,Summer,Barcelona,Basketball,Basketball Men's Basketball,NaN
1,2,A Lamusi,M,23.0,170.0,60.0,China,CHN,2012 Summer,2012,Summer,London,Judo,Judo Men's Extra-Lightweight,NaN
2,3,Gunnar Nielsen Aaby,M,24.0,NaN,NaN,Denmark,DEN,1920 Summer,1920,Summer,Antwerpen,Football,Football Men's Football,NaN
3,4,Edgar Lindenau Aabye,M,34.0,NaN,NaN,Denmark/Sweden,DEN,1900 Summer,1900,Summer,Paris,Tug-Of-War,Tug-Of-War Men's Tug-Of-War,Gold
4,5,Christine Jacoba Aaftink,F,21.0,185.0,82.0,Netherlands,NED,1988 Winter,1988,Winter,Calgary,Speed Skating,Speed Skating Women's 500 metres,NaN


In [5]:
conn = sqlite3.connect('olympics.db')

athlete_events.to_sql('athlete_events', conn, if_exists='replace', index=False)
host_city_locations.to_sql('host_city_locations', conn, if_exists='replace', index=False)

print('Database created successfully.')

Database created successfully.


Question 1: How many records are in athlete_events?

In [6]:
query1 = '''
SELECT COUNT(*) AS total_records
FROM athlete_events;
'''

pd.read_sql_query(query1, conn)

,total_records
0,271116


There are 271,116 records in the athlete_events table.

Question 2: How many different athletes competed in Olympics held in London?

In [7]:
query2 = '''
SELECT COUNT(DISTINCT ID) AS london_athletes
FROM athlete_events
WHERE City = 'London';
'''

pd.read_sql_query(query2, conn)

,london_athletes
0,16941


16,941 different athletes competed in Olympics held in London.

Question 3: Which Olympics host city is furthest north?

In [8]:
query3 = '''
SELECT City, lat
FROM host_city_locations
WHERE CAST(lat AS REAL) = (
    SELECT MAX(CAST(lat AS REAL))
    FROM host_city_locations
);
'''

pd.read_sql_query(query3, conn)

,City,lat
0,Lillehammer,61.115271


Lillehammer is the Olympics host city that is furthest north.

Question 4: What is the list of host cities with no duplicates?

In [9]:
query4 = '''
SELECT DISTINCT City
FROM host_city_locations
ORDER BY City;
'''

host_cities = pd.read_sql_query(query4, conn)
host_cities

,City
0,Albertville
1,Amsterdam
2,Antwerpen
3,Athina
4,Atlanta
5,Barcelona
6,Beijing
7,Berlin
8,Calgary
9,Chamonix


In [10]:
host_cities.to_csv('host_cities_FarahMakkawi.csv', index=False)
print('host_cities_FarahMakkawi.csv saved successfully.')

host_cities_FarahMakkawi.csv saved successfully.


In [11]:
files.download('host_cities_FarahMakkawi.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [12]:
conn.close()

Final answers

You can type these in markdown/text cells in the notebook:

How many records are in athlete_events?
271,116

How many different athletes competed in Olympics held in London?
16,941

Which Olympics host city is furthest north?
Lillehammer

List of host cities with no duplicates
Exported as host_cities_FarahMakkawi.csv